### DISCUSSION TITLE
Cannot run with mediapipe lib

### DISCUSSION THREAD

**Discussion by lazyfatcat** [beeware/briefcase#597] [Q&A]
just create a helloworld project，then add mediapipe lib，import mediapipe in app.py，it can build but cannot run app.  the following is test code
"""
My first application
"""
import toga
from toga.style import Pack
from toga.style.pack import COLUMN, ROW
# import cv2
import mediapipe as mp

class HelloWorld(toga.App):

    def startup(self):
        """
        Construct and show the Toga application.

        Usually, you would add your application to a main content box.
        We then create a main window (with a name matching the app), and
        show the main window.
        """
        main_box = toga.Box()

        self.main_window = toga.MainWindow(title=self.formal_name)
        self.main_window.content = main_box
        self.main_window.show()

def main():
    return HelloWorld(

**Comment by freakboy3742** [comment]
Thanks for the question.

In this case, the issue is almost certainly that you haven't declared the dependency on `mediapipe` in your `project.toml` requirements. Briefcase creates a completely standalone environment for your app. This means it's not enough to just `pip install mediapipe` in your development environment; you need to add it to `pyproject.toml` as well.

If you *have* added mediapipe to your `pyproject.toml`, then you may have missed running `briefcase update -d` to update the dependencies in the app.

If the app still doesn't work, there may be something else going on. However, to diagnose the problem, we will need a lot more detail. When you opened this bug report, you should have been prompted for a lot of detail about your environment - operating system, Python version, target platform, and more. We ask for this information for a reason - it's *essential* that you provide it, because it's a key part of diagnosing any problem that has occurred.

**Comment by lazyfatcat** [comment]
thanks for your reply, @freakboy3742 .
I have added mediapipe to pyproject.toml，and running briefcase build from the beginning, and the app cannot run(just crash and quit).
my environment is as follow:
1. macOS Big Sur 11.1
2. python 3.8.8
3. target platform is MacOS
4. running build run in terminal, it report error as follow
(env) ➜  helloworld git:(develop) ✗ briefcase build

[helloworld] Generating application template...
Using app template: https://github.com/beeware/briefcase-macOS-app-template.git, branch 3.8
Using existing template (sha cd859d950c109dd41c68f981338c8d48c5f709a7, updated Sun Oct 25 08:30:58 2020)

[helloworld] Installing support package...
Using support package https://briefcase-support.org/python?platform=macOS&version=3.8
... using most recent revision
Python-3.8-macOS-support.b4.tar.gz already downloaded
Unpacking support package...

[helloworld] Installing dependencies...
Collecting pyside2==5.13.0
  Using cached PySide2-5.13.0-5.13.0-cp35.cp36.cp37-abi3-macosx_10_12_intel.whl (119.9 MB)
Collecting mediapipe==0.8.2
  Using cached mediapipe-0.8.2-cp38-cp38-macosx_10_15_x86_64.whl (35.7 MB)
Collecting opencv-python==4.5.1.48
  Using cached opencv_python-4.5.1.48-cp38-cp38-macosx_10_13_x86_64.whl (40.3 MB)
Collecting dataclasses
  Using cached dataclasses-0.6-py3-none-any.whl (14 kB)
Collecting numpy==1.19.3
  Using cached numpy-1.19.3-cp38-cp38-macosx_10_9_x86_64.whl (15.9 MB)
Collecting six
  Using cached six-1.15.0-py2.py3-none-any.whl (10 kB)
Collecting absl-py
  Using cached absl_py-0.11.0-py3-none-any.whl (127 kB)
Collecting protobuf>=3.11.4
  Using cached protobuf-3.15.3-cp38-cp38-macosx_10_9_x86_64.whl (1.0 MB)
Collecting wheel
  Using cached wheel-0.36.2-py2.py3-none-any.whl (35 kB)
Collecting shiboken2==5.13.0
  Using cached shiboken2-5.13.0-5.13.0-cp35.cp36.cp37-abi3-macosx_10_12_intel.whl (775 kB)
Installing collected packages: six, numpy, wheel, shiboken2, protobuf, opencv-python, dataclasses, absl-py, pyside2, mediapipe
Successfully installed absl-py-0.11.0 dataclasses-0.6 mediapipe-0.8.2 numpy-1.19.3 opencv-python-4.5.1.48 protobuf-3.15.3 pyside2-5.13.0 shiboken2-5.13.0 six-1.15.0 wheel-0.36.2

[helloworld] Installing application code...
Installing src/helloworld...

[helloworld] Installing application resources...
Installing src/helloworld/resources/helloworld.icns as application icon...

[helloworld] Created macOS/Hello World

[helloworld] Built macOS/Hello World/Hello World.app
(env) ➜  helloworld git:(develop) ✗ briefcase build

[helloworld] Built macOS/Hello World/Hello World.app
(env) ➜  helloworld git:(develop) ✗ briefcase run  

[helloworld] Starting app...

The application cannot be opened for an unexpected reason, error=Error Domain=NSOSStatusErrorDomain Code=-10827 "kLSNoExecutableErr: The executable is missing" UserInfo={_LSLine=3665, _LSFunction=_LSOpenStuffCallLocal}

Unable to start app helloworld.
